tests/solana_tests/primitives.rs

```diff
@@ -1060,7 +1061,10 @@ fn test_mul_within_range_signed() {
         let mut contract = build_solidity(&src);
 
         // The range of values that can be held in signed N bits is [-2^(N-1), 2^(N-1)-1]. Here we generate a random number within this range and multiply it by -1, 1 or 0.
-        let first_operand_rand = rng.gen_bigint(width - 1).sub(1_u32);
+        let base = BigInt::from(2).pow(width - 1);
+        let lower_bound = base.clone().mul(-1);
+        let upper_bound = base.sub(1);
```

Comment:
The upper bound is exclusive https://docs.rs/num-bigint/latest/num_bigint/trait.RandBigInt.html#tymethod.gen_bigint_range

Reply:
Fixed.